infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -290,23 +302,113 @@ public void reboot() {
                 .<LogAddress,LogData>weigher((k, v) -> v.getData() == null ? 1 : v.getData().readableBytes())
                 .maximumWeight(maxCacheSize)
                 .removalListener(this::handleEviction)
-                .writer(new CacheWriter<LogAddress, LogData>() {
```

Comment:
@no2chem by the way this is the old pull request, essentially we want this :  
https://github.com/CorfuDB/CorfuDB/pull/353

I'll add the async logic in the handler.